src/pulselistener/pulselistener/lib/bus.py

```diff
@@ -46,7 +46,8 @@ def is_full(self):
         if isinstance(queue, asyncio.Queue):
             await queue.put(payload)
         else:
-            queue.put(payload)
+            # Run the synchronous mp queue.put in the asynchronous loop
+            await asyncio.get_running_loop().run_in_executor(None, lambda: queue.put(payload))
```

Comment:
This is a non-blocking call at the moment as the queue is unbounded, but it's better to be safe than sorry.